### PR TITLE
enforce UTF-8 gemfile parsing encoding #7

### LINKED ISSUE #7: Has trouble with non-ASCII character in old version of Nokogiri
Given a Gemfile of

    gem 'nokogiri', '1.5.11'
    gem 'annotate_gem', '~> 0.0.11'

The annotation has nonsensical letters instead of 鋸 (the kanji for nokogiri). I got

    # Nokogiri (é¸) is an HTML, XML, SAX, and Reader parser (http://nokogiri.org)

It works fine with nokogiri 1.6.0 .

Ruby version:

    ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin14]

Gemfile.lock:

    GEM
      specs:
        annotate_gem (0.0.11)
          bundler (~> 1.1)
        nokogiri (1.5.11)

    PLATFORMS
      ruby

    DEPENDENCIES
      annotate_gem (~> 0.0.11)
      nokogiri (= 1.5.11)

    BUNDLED WITH
       1.14.3

This might be an error in Nokogiri rather than annotate_gem: https://rubygems.org/gems/nokogiri/versions/1.5.11 also has a broken version of the character, while https://rubygems.org/gems/nokogiri/versions/1.6.0.rc1 is non-broken.

https://rubygems.org/gems/nokogiri/versions/1.5.1 has a non-broken character, while https://rubygems.org/gems/nokogiri/versions/1.5.2 has a broken character.

### PR DESCRIPTION
**Pull request by pocheptsov**
This PR fixes non-ascii symbol parsing in Gemfile. Also, had to fix travis build by upgrading supported ruby versions and some tricks with bundler installation. Fixes #7

### CODE REVIEW

File: lib/annotate_gem/gemfile.rb [hunk 1]
```diff
@@ -13,9 +13,10 @@ def initialize(gemfile_path, options = {})
     end
 
     def parse
-      self.source = File.readlines(gemfile_path)
+      self.source = File.readlines(gemfile_path, :encoding => 'UTF-8')
       source.each_with_index do |line, i|
-        if match = GEM_LINE_REGEX.match(line)
```
Comment (pocheptsov) [inline lib/annotate_gem/gemfile.rb]: Best practice to split assignment and condition check

**Comment by ivantsepp** [comment]
Hey! Thanks for your contribution! Sorry for the delay as I haven't been dabbling in Ruby for a while. Your code looks good so imma go ahead and merge!

**Comment by pocheptsov** [comment]
Would you mind to publish a new gem version? Thx Ivan.

**Comment by ivantsepp** [comment]
Just published https://rubygems.org/gems/annotate_gem/versions/0.0.12!

**Pull request merged**